### PR TITLE
EZP-28169 - Behat Covering of Sections

### PR DESCRIPTION
**Pull request by m-tyrala**
| Question      | Answer
| ------------- | ---
| Tickets       | [EZP-28169](https://jira.ez.no/browse/EZP-28169)
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes/no
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

13 tests that covers Admin -> Sections 

#### Checklist:
- [ ] Coding standards (`$ composer fix-cs`)
- [ ] Ready for Code Review

### CODE REVIEW

File: src/lib/Behat/PageObject/SectionPage.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    /**
+     * Verifies if list of Content items is empty.
+     *
+     * @param string $tabName
+     */
+    public function verifyListIsEmpty(): void
+    {
+        $firstRowValue = $this->adminLists['Content items']->table->getCellValue(1,1);
```
Comment (mnocon) [inline src/lib/Behat/PageObject/SectionPage.php]: I'd change it into two separate assertions:
Assert item count is equal to 1
Assert the message

File: src/lib/Behat/BusinessContext/AdministrationContext.php [hunk 2]
```diff
@@ -176,18 +177,37 @@ public function iGoToListItem(string $itemName, string $itemType, string $itemCo
      */
     public function iStartEditingItem(string $itemType, string $itemName, ?string $containerName = null): void
```
Comment (mnocon) [inline src/lib/Behat/BusinessContext/AdministrationContext.php]: I think this one should be split into two separate steps:
One:
```
* @When I start editing :itemType :itemName		     
* @When I start editing :itemType :itemName from :containerName
```
and second
```
 
* @When I start editing :itemType :itemName from details page
```

File: src/lib/Behat/BusinessContext/AdministrationContext.php [hunk 3]
```diff
@@ -176,18 +177,37 @@ public function iGoToListItem(string $itemName, string $itemType, string $itemCo
...
+            ->startEditingItem($itemName);
     }
 
     /**
-     * @When I delete :itemType :itemName
-     * @When I delete :itemType :itemName from :itemContainer
+     * @When I delete :itemType
+     * @When I delete :itemType from :containerName
      */
-    public function iDeleteItem(string $itemType, string $itemName, ?string $itemContainer = null): void
```
Comment (mnocon) [inline src/lib/Behat/BusinessContext/AdministrationContext.php]: Same here: I'd split it into 3 steps (I delete, I delete from container, I delete from details page). I don't think we need to have such complicated steps.

File: features/Sections.feature [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  @javascript @common
+  Scenario: Content item can be reassigned to section from the Sections details
+    Given I go to "Media" "Section" page
+    When I start assigning to "Media" from "Section" page
+      And I select content "Media/Images" through UDW
+      And I confirm UDW the selection
+    Then I should be on "Section" "Media" page
+      And content items list in section "Media" contains items
+      | Name   | Content Type | Path  |
+      | Images | Folder       | Media |
```
[the remark this answers is not in the document]
Comment (m-tyrala) [reply features/Sections.feature]: Next scenario `Empty section can be deleted` serves as an assertion here. I don't really want to create `Then` steps that contains navigation actions.

[the remark this answers is not in the document]
Comment (mnocon) [reply features/Sections.feature]: You're right that it's not pretty (I don't have any better idea). But still, for me it's better to have a clunky assertion in the same test which does the action than to have a hidden assertion in the next test (I didn't know that Sections behave like that and was really puzzled what happened to the assigned item 🤔 )

One potential reason: one day we may decide that the one of these tests is less important and we will run it less often - meaning that this assertion will also be executed less often.

**Comment by m-tyrala** [comment]
@lserwatka this one can be merged

**Pull request merged**